WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginEmailFragment.java

```diff
@@ -0,0 +1,397 @@
+import android.view.View;
+import android.view.View.OnClickListener;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import org.greenrobot.eventbus.Subscribe;
+import org.greenrobot.eventbus.ThreadMode;
+import org.wordpress.android.BuildConfig;
+import org.wordpress.android.R;
+import org.wordpress.android.WordPress;
+import org.wordpress.android.fluxc.Dispatcher;
+import org.wordpress.android.fluxc.generated.AccountActionBuilder;
+import org.wordpress.android.fluxc.store.AccountStore.OnAvailabilityChecked;
+import org.wordpress.android.ui.accounts.AbstractFragment;
+import org.wordpress.android.ui.accounts.SignInDialogFragment;
+import org.wordpress.android.ui.accounts.login.nav.LoginNav;
+import org.wordpress.android.ui.accounts.login.nav.LoginStateGetter;
+import org.wordpress.android.util.AppLog;
+import org.wordpress.android.util.AppLog.T;
+import org.wordpress.android.util.EditTextUtils;
+import org.wordpress.android.util.NetworkUtils;
+import org.wordpress.emailchecker2.EmailChecker;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.inject.Inject;
+
+public class LoginEmailFragment extends AbstractFragment implements TextWatcher {
+    private static final String KEY_IN_PROGRESS = "KEY_IN_PROGRESS";
+
+    public static final String TAG = "login_email_fragment_tag";
+    public static final int MAX_EMAIL_LENGTH = 100;
+
+    protected TextInputLayout mEmailEditTextLayout;
+    protected EditText mEmailEditText;
+
+    protected boolean mEmailAutoCorrected;
+
+    protected Button mNextButton;
+    protected View mUsernamePasswordButton;
+
+    protected @Inject Dispatcher mDispatcher;
+
+    private LoginNav.InputEmail mLoginNavInputEmail;
+
+    private ProgressDialog mProgressDialog;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        ((WordPress) getActivity().getApplication()).component().inject(this);
+
+        setHasOptionsMenu(true);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        ViewGroup rootView = (ViewGroup) inflater.inflate(R.layout.login_email_screen, container, false);
+
+        mEmailEditText = (EditText) rootView.findViewById(R.id.login_email);
+        mEmailEditText.addTextChangedListener(this);
+
+        mEmailEditTextLayout = (TextInputLayout) rootView.findViewById(R.id.login_email_layout);
+
+        mNextButton = (Button) rootView.findViewById(R.id.login_email_next_button);
+        mNextButton.setOnClickListener(mNextClickListener);
+        mUsernamePasswordButton = rootView.findViewById(R.id.login_email_username_password);
+
+        mEmailEditText.setOnFocusChangeListener(new View.OnFocusChangeListener() {
+            public void onFocusChange(View v, boolean hasFocus) {
+                if (!hasFocus) {
```

Comment:
Hey @mattmiklic cool, so, have a look at the production app and try these steps to trigger the email autocorrect:
* be logged out of wpcom and go to login
* in the username/email field hit Enter to reveal the password field. This is useful for triggering the "lose-focus" event down the road.
* type in this email: `a@b.com` and hit the Enter again. This will move the input focus to the password field but in the same time it will auto-change the email to `a@bt.com`.

The email autocorrect tries to automatically fix some "common" typos in the domain names part of the email address. The current UI does the fix silently and instantly. If the user tries to re-edit the email (perhaps to put it back to `a@b.com`) the app respects it and doesn't try to auto-correct again for that apo session.

I agree with you that this falls into the "don't be too clever" territory. It either needs more work to make it more clear to the user about what happened there or drop it alltogether. Have a look and share your thoughts here, thanks!